### PR TITLE
[fix][doc] Clarifying non-persistent topics a bit more

### PR DESCRIPTION
**Pull request by asafm**
### Modifications

* Clarify the notion that non-persistent topics do not buffer in-memory messages at all.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `no-need-doc` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-added`
(Docs have been already added)

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]

**Review by HQebupt — APPROVED** [review]

**Review by Anonymitaet — COMMENTED** [review]
Thanks for your contribution!

Does this apply to other doc versions? If so, would you like to update them?

**Comment by Anonymitaet** [comment]
> despair

@asafm just `batch and replace` in one click

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -813,12 +812,14 @@ non-persistent://tenant/namespace/topic
...
 
 By default, non-persistent topics are enabled on Pulsar brokers. You can disable them in the broker's [configuration](reference-configuration.md#broker-enableNonPersistentTopics). You can manage non-persistent topics using the `pulsar-admin topics` command. For more information, see [`pulsar-admin`](https://pulsar.apache.org/tools/pulsar-admin/).
 
+Currently non-persistent topics which are not partitioned are not persisted to ZK, which means if the broker owning them crashes, they won't get re-assigned to another broker because they only existed in the owner broker memory. The current work-around for that is to make sure `allowAutoTopicCreation` broker config is `true` and `allowAutoTopicCreationType` is `non-partitioned` (Those the the default values).
```
[the remark this answers is not in the document]
Comment (Anonymitaet) [reply site2/docs/concepts-messaging.md:821]: do not use `ZK` for ZooKeeper`, spell it out
guide: https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.zzzwnxsvrwb

[the remark this answers is not in the document]
Comment (asafm) [reply site2/docs/concepts-messaging.md]: @Anonymitaet fixed

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**